src/app/tests/suites/certification/Test_TC_DISHM_2_1.yaml

```diff
@@ -80,106 +80,273 @@ tests:
+              - name: "expectedValue"
+                value: "y"
 
     - label:
-          "Step 4: Manually put the device in a state from which it will FAIL to
+          "Manually put the device in a state from which it will FAIL to
           transition to PIXIT.DISHM.MODE_CHANGE_FAIL"
-      verification: |
-          ChangeToModeWithStatus Command is not implented.
-      disabled: true
+      cluster: "LogCommands"
+      command: "UserPrompt"
+      PICS: PICS_USER_PROMPT
+      arguments:
+          values:
+              - name: "message"
+                value: "Please enter 'y' for success"
+              - name: "expectedValue"
+                value: "y"
 
-    - label: "Step 5: TH reads from the DUT the CurrentMode attribute."
-      PICS: DISHM.S.A0001
+    - label: "TH reads the CurrentMode attribute from the DUT"
       verification: |
-          ./chip-tool dishwashermodeselect read current-mode 1 1
+          ./chip-tool dishwashermode read current-mode 1 1
 
-          Verify on TH(chip-tool) logs, CurrentMode attribute value is an integer value and equal to the new_mode_th, below is the sample log provided for the raspi platform:
+          Verify on TH(chip-tool) logs, CurrentMode attribute value is an integer value
+          - Save the value as old_current_mode_dut,
+          below is the sample log provided for the raspi platform:
 
           [1685018283.726571][16418:16420] CHIP:DMG: SuppressResponse = true,
           [1685018283.726581][16418:16420] CHIP:DMG: InteractionModelRevision = 1
           [1685018283.726589][16418:16420] CHIP:DMG: }
           [1685018283.726667][16418:16420] CHIP:TOO: Endpoint: 1 Cluster: 0x0000_0050 Attribute 0x0000_0003 DataVersion: 3898153349
-          [1685018283.726680][16418:16420] CHIP:TOO: CurrentMode: 0
-      disabled: true
+          [1685018283.726680][16418:16420] CHIP:TOO: CurrentMode: 2
+      cluster: "LogCommands"
+      command: "UserPrompt"
+      PICS: PICS_USER_PROMPT
+      arguments:
+          values:
+              - name: "message"
+                value: "Please enter 'y' for success"
+              - name: "expectedValue"
+                value: "y"
 
     - label:
-          "Step 6: TH sends a ChangeToMode command to the DUT with NewMode set
-          to PIXIT.DISHM.MODE_CHANGE_FAIL"
-      PICS: DISHM.S.C00.Rsp
+          "TH sends a ChangeToMode command to the DUT with NewMode set to
+          PIXIT.DISHM.MODE_CHANGE_FAIL"
       verification: |
-          ChangeToModeWithStatus Command is not implented.
-      disabled: true
+          ./chip-tool dishwashermode change-to-mode 2 1 1
 
-    - label: "Step 7: TH reads from the DUT the CurrentMode attribute."
-      PICS: DISHM.S.A0001
-      verification: |
+          On TH(chip-tool) log, verify DUT responds with a GenericFailure (value 0x02) or MfgCodes (0x80 to 0xBF range) status response and below is the sample log provided for the raspi platform:
```

Comment:
The test plan has additional conditions around the StatusText field, but those conditions do not match the spec.  Please file a test plan issue and reference it here.